src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php

```diff
@@ -67,7 +68,7 @@ public function loadClassMetadata(ClassMetadata $metadata)
                                 $metadata->addAttributeGroup(lcfirst($matches[2]), $group);
                             }
                         } else {
-                            throw new \BadMethodCallException(sprintf('Groups on "%s::%s" cannot be added. Groups can only be added on methods beginning with "get" or "is".', $className, $method->name));
+                            throw new MappingException(sprintf('Groups on "%s::%s" cannot be added. Groups can only be added on methods beginning with "get" or "is".', $className, $method->name));
```

Comment:
I was thinking about the same thing. Another argument in favor of using a `MappingException`: it's what the Validator Annotation Loader throws https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Validator/Mapping/Loader/AnnotationLoader.php#L84